audittools/backing_store.go

```diff
@@ -0,0 +1,47 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company
+// SPDX-License-Identifier: Apache-2.0
+
+package audittools
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sapcc/go-api-declarations/cadf"
+)
+
+// BackingStore is an interface for buffering audit events when the primary sink (RabbitMQ) is unavailable.
+// This prevents audit data loss during RabbitMQ outages or network disruptions.
+//
+// Thread safety requirements vary by implementation - see specific implementation documentation.
+type BackingStore interface {
+	// Init initializes the backing store with the given Prometheus registry.
```

Comment:
The phrasing of "with the given Prometheus registry" is vague:
```suggestion
	// Init initializes the backing store.
	// If the implementation wants to define its own Prometheus metrics, it shall use the provided registry.
```